app/models/artefact.rb

```diff
@@ -318,6 +318,16 @@ def save_as(user, options={})
     save(options)
   end
 
+  # performed tasks have no user associated
```

Comment:
Perhaps it's useful here to explain that we should use this task when doing save actions from rake tasks, message queue consumers, etc.